File: packages/web/src/features/student/studentPresentation.ts
import type { Student } from '@sprout/shared';
import { gradeLevelToBand } from '../learn/gradeBand';

/** Whether a student sees the collapsed, early-reader `Today` presentation
 * (short copy, icon+text, optional read-aloud, one scrollable view) rather
 * than the standard four-destination Today/History/Goals/Learn nav — see
 * 01_EXPERIENCE_FOUNDATIONS.md §5.2: "Pre-K–2 may collapse these into one
 * scrollable Today view." Derived from the student's own roster
 * `gradeLevel` (staff-authorized), never a self-reported age — see
 * gradeBand.ts. An unresolvable grade level fails closed to the *standard*
 * presentation (the fuller, more capable one), not the simplified one —
 * unlike Learn's "ask an adult" fail-closed, hiding functionality here has
 * no safety upside and would just make the app harder to use for a
 * student whose grade level wasn't recorded correctly. */
export function isEarlyReaderPresentation(student: Pick<Student, 'gradeLevel'>): boolean {
  const band = gradeLevelToBand(student.gradeLevel);
  return band === 'Pre-K–K' || band === 'Grades 1–2';
}

File: packages/mobile/lib/features/student/student_presentation.dart
import 'package:sprout/core/models/student.dart';
import 'package:sprout/features/learn/grade_band.dart';

/// Dart mirror of packages/web/src/features/student/studentPresentation.ts
/// — see that file's doc comment for the full reasoning (fails closed to
/// the standard, fuller presentation on an unresolvable grade level, not
/// the simplified one).
bool isEarlyReaderPresentation(Student student) {
  final band = gradeLevelToBand(student.gradeLevel);
  return band == 'Pre-K–K' || band == 'Grades 1–2';
}
